### PR TITLE
Docs: Update rule descriptions (fixes #5912)

### LINKED ISSUE #5912: Update rule descriptions
This is just a small tracking issue for the few rules that specifically refer to a kwyrod (`var` and `function`) in their descriptions even though the rules apply to the broader concept (`let` and/or `const` and arrow functions). As the metadata is added to the rules, these can be updated then, and the remaining descriptions can be updated when the metadata updates are complete. The rules that only apply to specific keywords do not need to be updated.

The descriptions to be updated are in https://github.com/eslint/eslint/pull/5890. Specifically:

 - [x] `func-call-spacing`: require or disallow spacing between function identifiers and their invocations
 - [x] `init-declarations`: require or disallow initialization in variable declarations
 - [x] `max-params`: enforce a maximum number of parameters in function definitions
 - [x] `max-statements`: enforce a maximum number of statements allowed in function blocks
 - [x] `new-cap`: require constructor names to begin with a capital letter
 - [x] `newline-after-var`: require or disallow an empty line after variable declarations
 - [x] `no-implicit-globals`: disallow variable and `function` declarations in the global scope
 - [x] `no-inner-declarations`: disallow variable or `function` declarations in nested blocks (recommended)
 - [x] `no-mixed-requires`: disallow `require` calls to be mixed with regular variable declarations
 - [x] `no-redeclare`: disallow variable redeclaration (recommended)
 - [x] `no-shadow`: disallow variable declarations from shadowing variables declared in the outer scope
 - [x] `no-spaced-func`: disallow spacing between function identifiers and their applications
 - [x] `one-var-declaration-per-line`: require or disallow newlines around variable declarations
 - [x] `sort-keys`: require object keys to be sorted
 - [x] `space-infix-ops`: require spacing around infix operators

### PR DESCRIPTION
**Pull request by scriptdaemon**
**What is the purpose of this pull request? (put an "X" next to item)**

 - [x] Documentation update
 - [ ] Bug fix ([template](https://github.com/eslint/eslint/blob/master/templates/bug-report.md))
 - [ ] New rule ([template](https://github.com/eslint/eslint/blob/master/templates/rule-proposal.md))
 - [ ] Changes an existing rule ([template](https://github.com/eslint/eslint/blob/master/templates/rule-change-proposal.md))
 - [ ] Add autofixing to a rule
 - [ ] Add a CLI option
 - [ ] Add something to the core
 - [ ] Other, please explain:

*If the item you've check above has a template, please paste the template questions here and answer them.*

**Please check each item to ensure your pull request is ready:**

- [x] I've read the [pull request guide](http://eslint.org/docs/developer-guide/contributing/pull-requests)
- [ ] I've included tests for my change
- [x] I've updated documentation for my change (if appropriate)

**What changes did you make? (Give an overview)**

Updated confusing / incorrect rule descriptions that refer to language keywords that instead apply to the overall concept (e.g. rules that apply to `const`, `let`, and `var`, but only include `var` in the description).

**Is there anything you'd like reviewers to focus on?**

Does the grammar make sense? Are there other rules that should be added to this?

The issue this PR references has not been accepted, but I thought I'd get the ball rolling.

### CODE REVIEW

File: lib/rules/no-implicit-globals.js [hunk 1]
```diff
@@ -12,7 +12,7 @@
 module.exports = {
     meta: {
         docs: {
-            description: "disallow `var` and named `function` declarations in the global scope",
+            description: "disallow variable and `function` declarations in the global scope",
```
Comment (kaicataldo) [inline lib/rules/no-implicit-globals.js]: Is there a reason why the backticks aren't removed from function here?

**Comment by kaicataldo** [comment]
Do we also want to update the titles of corresponding docs files while we're at it?

**Comment by scriptdaemon** [comment]
@kaicataldo Yes, exactly. We had intended to do both at the same time, but that portion slipped my mind. Update incoming.

**Comment by nzakas** [comment]
Nice! Thanks @scriptdaemon

**Pull request merged**